docs: misc (#32530)
neovim/neovim: .github/workflows/notes.md

### Linux (x86_64)

> If your glibc version is too low, try the (unsupported) [builds for older glibc](https://github.com/neovim/neovim-releases).

Reviewer: ```suggestion
If your system does not have the [required glibc version](https://neovim.io/doc/user/support.html#_supported-platforms), try the (unsupported) [builds for older glibc](https://github.com/neovim/neovim-releases).
```

> If your system does not have the [required glibc version](https://neovim.io/doc/user/support.html#_supported-platforms), try the (unsupported) [builds for older glibc](https://github.com/neovim/neovim-releases).

Reviewer: sorry, a small URL improvement (both work, but this one is more "correct" and less risk of breaking if the ToC logic changes in the future)

```suggestion
If your system does not have the [required glibc version](https://neovim.io/doc/user/support.html#supported-platforms), try the (unsupported) [builds for older glibc](https://github.com/neovim/neovim-releases).
```
